pkg/kudoctl/setup/prereqs.go

```diff
@@ -0,0 +1,69 @@
+package setup
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kube"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+// Defines a single prerequisite that is defined as a k8s resource
+type k8sResource interface {
+	Install(client *kube.Client) error
+	Validate(client *kube.Client) error
+	AsRuntimeObj() []runtime.Object
+}
```

Comment:
This is one of the nicer things of the refactoring: There's an abstraction for one or more manifests as a prerequisite, currently implemented in each of the `prereq_*.go` files. We can extend that with methods for migration, deletion, etc.